### ISSUE TITLE
Queries do not work correctly after the index is created

### ISSUE THREAD

**Issue by jfontan** [src-d/go-mysql-server#238]
Queries with the indexed expression do not return correct answers. For example:

```
MySQL [(none)]> select file_path, language(file_path, blob_content) from files;
+-----------------------------+-----------------------------------------------+
| file_path                   | language(files.file_path, files.blob_content) |
+-----------------------------+-----------------------------------------------+
| executor.go                 | Go                                            |
| executor_test.go            | Go                                            |
| pack.go                     | Go                                            |
| pack_test.go                | Go                                            |
| server.go                   | Go                                            |
| server_test.go              | Go                                            |
| borges.go                   | Go                                            |
...
| sivafuse/get_path.go        | Go                                            |
| sivafuse/get_path_test.go   | Go                                            |
| sivafuse/git.go             | Go                                            |
| sivafuse/root_siva_fs.go    | Go                                            |
+-----------------------------+-----------------------------------------------+
681 rows in set (0.94 sec)
```

Query with the indexed expression:

```
MySQL [(none)]> select file_path from files where language(file_path, blob_content) = 'Go';
Empty set (0.02 sec)
```

The index was created with:

```
create index language on files using pilosa (language(file_path, blob_content));
```

I've double checked that the index was already created. To be sure that no leftovers were there I've also started with new index directories and started clean pilosa servers.

**Comment by kuba--** [comment]
Root cause:
When we create a pilosa index we call `Create` with one expression:
`language(files.file_path, files.blob_content)` what gives ExpressionHash `83700d6ec9d5a296ba3231e4bd4498786fb9473e`

But then we call `Save` with ` sql.IndexKeyValueIter` which for ` iter.Next()` returns slice of values (file_path, blob_content) instead of one value as a result of `language(files.file_path, files.blob_content)`.

In other words we indexed file paths instead of language names (that's why we cannot find 'Go' or any other language).

For single column indexes everything works fine, e.g.:
```
CREATE INDEX idx_file ON files USING pilosa (file_path);

SELECT * FROM files WHERE file_path = 'LICENSE';
```

**Comment by kuba--** [comment]
Close this one after upgrade mysql
